$OpenBSD: patch-gio_tests_file_c,v 1.6 2013/09/25 12:37:07 ajacoutot Exp $

XXX why isn't this working with kqueue?
push upstream when fixed

--- gio/tests/file.c.orig	Thu Aug  8 12:00:40 2013
+++ gio/tests/file.c	Tue Sep 24 10:05:43 2013
@@ -450,11 +450,12 @@ test_create_delete (gconstpointer d)
   data->monitor = g_file_monitor_file (data->file, 0, NULL, &error);
   g_assert_no_error (error);
 
-  /* This test doesn't work with GPollFileMonitor, because it assumes
-   * that the monitor will notice a create immediately followed by a
-   * delete, rather than coalescing them into nothing.
+  /* This test doesn't work with GPollFileMonitor nor GKqueueFileMonitor,
+   * because it assumes that the monitor will notice a create immediately
+   * followed by a delete, rather than coalescing them into nothing.
    */
-  if (!strcmp (G_OBJECT_TYPE_NAME (data->monitor), "GPollFileMonitor"))
+  if ((!strcmp (G_OBJECT_TYPE_NAME (data->monitor), "GPollFileMonitor")) ||
+      (!strcmp (G_OBJECT_TYPE_NAME (data->monitor), "GKqueueFileMonitor")))
     {
       g_print ("skipping test for this GFileMonitor implementation");
       goto skip;
